packages/web/src/store/tipping/sagas.ts

```diff
@@ -0,0 +1,125 @@
+import BN from 'bn.js'
+import { call, put, select, takeEvery } from 'typed-redux-saga/macro'
+
+import { Name } from 'common/models/Analytics'
+import { BNWei } from 'common/models/Wallet'
+import { FeatureFlags } from 'common/services/remote-config'
+import { getAccountUser } from 'common/store/account/selectors'
+import { transferEthAudioToSolWAudio } from 'common/store/pages/token-dashboard/slice'
+import { getSendTipData } from 'common/store/tipping/selectors'
+import {
+  confirmSendTip,
+  sendTipFailed,
+  sendTipSucceeded
+} from 'common/store/tipping/slice'
+import { getAccountBalance } from 'common/store/wallet/selectors'
+import { decreaseBalance } from 'common/store/wallet/slice'
+import { weiToAudioString, weiToString } from 'common/utils/wallet'
+import { remoteConfigInstance } from 'services/remote-config/remote-config-instance'
+import walletClient from 'services/wallet-client/WalletClient'
+import { make } from 'store/analytics/actions'
+
+const { getFeatureEnabled } = remoteConfigInstance
+
+function* sendTipAsync() {
+  const isTippingEnabled = getFeatureEnabled(FeatureFlags.TIPPING_ENABLED)
+  if (!isTippingEnabled) {
+    return
+  }
+
+  const account = yield* select(getAccountUser)
+  if (!account) {
+    return
+  }
+
+  const sendTipData = yield* select(getSendTipData)
+  const { user, amount: weiBNAmount } = sendTipData
+  if (!user) {
+    return
+  }
+
+  const recipientWallet = user.spl_wallet
+  const weiBNBalance: BNWei = yield select(getAccountBalance) ??
+    (new BN('0') as BNWei)
+  const waudioWeiAmount = yield* call(walletClient.getCurrentWAudioBalance)
+
+  if (weiBNAmount.gt(weiBNBalance)) {
+    const error = 'Not enough $AUDIO'
+    console.error(`Send tip failed: ${error}`)
+    yield put(sendTipFailed({ error }))
+    return
+  }
+
+  try {
+    yield put(
+      make(Name.TIP_AUDIO_REQUEST, {
+        senderWallet: account.spl_wallet,
+        recipientWallet,
+        senderHandle: account.handle,
+        recipientHandle: user.handle,
+        amount: weiToAudioString(weiBNAmount)
+      })
+    )
+    // If transferring spl wrapped audio and there are insufficent funds with only the
+    // user bank balance, transfer all eth AUDIO to spl wrapped audio
+    if (weiBNAmount.gt(waudioWeiAmount)) {
+      yield put(transferEthAudioToSolWAudio())
+      yield call(walletClient.transferTokensFromEthToSol)
+    }
```

Comment:
Worth having some extra state for this, maybe some some UI similar to what we have on the transfer audio modal explaining that this may take longer than usual? Let's check with design

Reply:
yup this is something I meant to bring up too